x/pricefeed/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,201 @@
+package keeper_test
+
+import (
+	"testing"
+	"time"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/kava-labs/kava/app"
+	"github.com/kava-labs/kava/x/pricefeed/keeper"
+	"github.com/kava-labs/kava/x/pricefeed/types"
+	"github.com/stretchr/testify/suite"
+	tmprototypes "github.com/tendermint/tendermint/proto/tendermint/types"
+)
+
+type grpcQueryTestSuite struct {
+	suite.Suite
+
+	tApp   app.TestApp
+	ctx    sdk.Context
+	keeper keeper.Keeper
+	addrs  []sdk.AccAddress
+	now    time.Time
+}
+
+func (suite *grpcQueryTestSuite) SetupTest() {
+	suite.tApp = app.NewTestApp()
+	suite.ctx = suite.tApp.NewContext(true, tmprototypes.Header{}).
+		WithBlockTime(time.Now().UTC())
+	suite.keeper = suite.tApp.GetPriceFeedKeeper()
+
+	_, addrs := app.GeneratePrivKeyAddressPairs(5)
+	suite.addrs = addrs
+
+	suite.now = time.Now().UTC()
+}
+
+func (suite *grpcQueryTestSuite) TestGrpcParams() {
+	tests := []struct {
+		giveMsg      string
+		giveParams   types.Params
+		wantAccepted bool
+	}{
+		{"default params", types.DefaultParams(), true},
+		{"test params", types.NewParams([]types.Market{
+			{MarketID: "tstusd", BaseAsset: "tst", QuoteAsset: "usd", Oracles: []string{}, Active: true},
+		}), true},
+	}
+
+	for _, tt := range tests {
+		suite.Run(tt.giveMsg, func() {
+			suite.keeper.SetParams(suite.ctx, tt.giveParams)
+
+			res, err := suite.keeper.Params(sdk.WrapSDKContext(suite.ctx), &types.QueryParamsRequest{})
+
+			if tt.wantAccepted {
+				suite.NoError(err)
+				suite.NoError(tt.giveParams.VerboseEqual(res.Params), "params query should respond with set params")
+			} else {
+				suite.Error(err)
+			}
+		})
+	}
+}
+
+func (suite *grpcQueryTestSuite) TestGrpcPrice() {
+	params := types.NewParams([]types.Market{
+		{MarketID: "tstusd", BaseAsset: "tst", QuoteAsset: "usd", Oracles: []string{}, Active: true},
+	})
+	suite.keeper.SetParams(suite.ctx, params)
+
+	// No prices set yet, should error
+	_, err := suite.keeper.Price(sdk.WrapSDKContext(suite.ctx), &types.QueryPriceRequest{MarketId: "tstusd"})
+	suite.ErrorIs(types.ErrNoValidPrice, err)
+
+	suite.setTstPrice()
+
+	expectedPrice := types.NewCurrentPrice("tstusd", sdk.MustNewDecFromStr("0.34"))
+
+	res, err := suite.keeper.Price(sdk.WrapSDKContext(suite.ctx), &types.QueryPriceRequest{MarketId: "tstusd"})
+	suite.NoError(err)
+	// Price median of addTstPrices below
+	suite.Equal(expectedPrice, res.Price)
+
+	_, err = suite.keeper.Price(sdk.WrapSDKContext(suite.ctx), &types.QueryPriceRequest{MarketId: "invalid"})
+	suite.Equal("rpc error: code = InvalidArgument desc = invalid market ID", err.Error())
+
+	prices, err := suite.keeper.Prices(sdk.WrapSDKContext(suite.ctx), &types.QueryPricesRequest{})
+	suite.NoError(err)
+
+	// Includes previous same price
+	suite.Contains(prices.Prices, expectedPrice)
+}
```

Comment:
nit: Looks like this test tests 4 different scenarios. In general I'd recommend splitting up tests like these to make it clear.
It makes it easier to for future devs to see what behvaiour is tested. And large tests tend to get larger and harder to read over time.

Could do separate tests like `TestGrpcPrice_NoPriceSet()`, or use separate `suite.Run(...`s within a single test method, or do a table driven test like in `kavadist/grpc_query_test.go`